mlir/lib/Analysis/Presburger/IntegerRelation.cpp

```diff
@@ -2498,6 +2499,33 @@ void IntegerRelation::printSpace(raw_ostream &os) const {
   os << getNumConstraints() << " constraints\n";
 }
 
+void IntegerRelation::removeTrivialEqualities() {
+  for (int i = getNumEqualities() - 1; i >= 0; --i)
+    if (rangeIsZero(getEquality(i)))
+      removeEquality(i);
+}
+
+bool IntegerRelation::isFullDim() {
+  if (getNumVars() == 0)
+    return true;
+  if (isEmpty())
+    return false;
+
+  // If there is a non-trivial equality, the space cannot be full-dimensional.
+  removeTrivialEqualities();
+  if (getNumEqualities() > 0)
+    return false;
+
+  // If along the direction of any of the inequalities, the upper and lower
+  // optima are the same, then the region is not full-dimensional.
+  Simplex simplex(*this);
+  bool fullDim = llvm::none_of(llvm::seq<int>(0, getNumInequalities()),
+                               [&](unsigned i) -> bool {
```

Comment:
you can skip the return type `->bool` as it is inferred

why is llvm:seq with `<int>` while the parameter is unsigned? I would just use int in both.

you can directly return the value, the `fullDim` variable doesn't seem to add anything here